documentation/boltspec_reference.md

```diff
@@ -0,0 +1,1117 @@
+# BoltSpec reference
+
+The `BoltSpec` helper library includes several functions to help write unit
+tests for plans. For more information about writing unit tests for plans, and
+how to use these functions, see [Testing plans](testing_plans.md).
+
+## Mocks
+
+Mocks serve two purposes when you are writing tests for your plans: they allow
+assertions about executor functions invoked during a plan run and also allow you
+to set their return values. For example, you might use a mock to make an
+assertion that the `run_task` function is invoked exactly one time during a
+test.
+
+### `expect_command`
```

Comment:
This page overall seems pretty verbose, and like it includes a lot of duplication of expectations. What do you think about having a separate `Expectations` page that lists all the expectations, and then each function can just list the expectations it accepts and link to the Expectations reference page?

Reply:
I don't think verbosity/length should be a concern for reference pages since they're not meant to read straight through and have a ToC for easy navigation. Some of the modifiers also have different behavior (`be_called_times`) or have different parameters (`always_return`, `return_for_targets`, `return`) depending on the mock/stub used. We _could_ enumerate the exceptions for each mock/stub under the modifier, but I think that would make the page more confusing than anything.

Reply:
Yeah, I agree. I think the way we expect users to interact with these reference pages is by using the TOC or using CMD+F, and it's reasonable for reference pages to be fairly repetitive for the sake of completeness. I'm also a little wary of adding more reference pages than we have to cos the references section is starting to get kinda chonky.